### PR TITLE
[UM-729] Update repository setup

### PR DESCRIPTION
**Pull request by kirill-panoply**
# Description

- Add CI/CD workflow
- Update project structure (move code into `src` and tests into `test`)
- Misc updates (`PULL_REQUEST_TEMPLATE.md`, `README.md`, `.gitignore`, `package.json`)

### CODE REVIEW

**Comment by kirill-panoply** [comment]
@ilia-panoply I don't have any permissions in this repository

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+on:
+  push:
+    branches: [master]
+
+  pull_request:
+    branches: [master]
+
+env:
+  node_version: 10.14
```
Comment (ilia-panoply) [inline .github/workflows/main.yml:11]: @kirill-panoply please update the version to node 14

File: package.json [hunk 2]
```diff
@@ -1,44 +1,25 @@
...
-    "tar": "^2.2.1",
-    "xlsx": "^0.8.2"
-  },
-  "dependencies": {
-    "overflow": "1.1.4"
-  }
+    "name": "structurize",
+    "version": "1.4.2",
+    "description": "Text format hinting library for estimating the format of a given input string",
+    "author": "Roi Avinoam <avinoamr@gmail.com>",
```
Comment (ilia-panoply) [inline package.json]: please change the author to panoply.io

**Review by ilia-panoply — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (kirill-panoply) [reply package.json]: Will this change meet [MIT license](https://github.com/panoplyio/structurize/blob/master/LICENSE) limitations and conditions?

File: .github/workflows/main.yml (hunk 1 above)
Comment (kirill-panoply) [reply .github/workflows/main.yml:11]: I don't think we should do it since we're using this package in applications which are running on node 10.14

Comment (ilia-panoply) [reply .github/workflows/main.yml:11]: as I mentioned in overflow PR, @OmriGLevy is working on node 14 migration

File: package.json (hunk 2 above)
Comment (ilia-panoply) [reply package.json]: no, and anyway we are not planing to publish this package on NPM

File: .github/workflows/main.yml (hunk 1 above)
Comment (kirill-panoply) [reply .github/workflows/main.yml:11]: Yeah so let's update node version here after it will be done.

File: package.json (hunk 2 above)
Comment (kirill-panoply) [reply package.json]: Yeah, but it's public repo and the author of this lib is indeed Roi Avinoam,

**Review by ilia-panoply — APPROVED** [review]

**Pull request merged**